editor/plugins/texture_editor_plugin.cpp

```diff
@@ -30,132 +30,60 @@
 
 #include "texture_editor_plugin.h"
 
-#include "core/config/project_settings.h"
-#include "core/io/resource_loader.h"
-#include "editor/editor_settings.h"
+TexturePreview::TexturePreview(Ref<Texture2D> p_texture, bool p_show_metadata) {
+	Ref<Theme> theme = EditorNode::get_singleton()->get_editor_theme();
 
-void TextureEditor::_gui_input(Ref<InputEvent> p_event) {
-}
-
-void TextureEditor::_notification(int p_what) {
-	if (p_what == NOTIFICATION_READY) {
-		//get_scene()->connect("node_removed",this,"_node_removed");
-	}
-
-	if (p_what == NOTIFICATION_DRAW) {
-		Ref<Texture2D> checkerboard = get_theme_icon("Checkerboard", "EditorIcons");
-		Size2 size = get_size();
+	TextureRect *checkerboard = memnew(TextureRect);
+	checkerboard->set_texture(theme->get_icon("Checkerboard", "EditorIcons"));
+	checkerboard->set_stretch_mode(TextureRect::STRETCH_TILE);
+	checkerboard->set_custom_minimum_size(Size2(0.0, 256.0));
```

Comment:
Scale for hiDPI displays:

```suggestion
	checkerboard->set_custom_minimum_size(Size2(0.0, 256.0) * EDSCALE);
```